src/main/java/org/opensearch/testcontainers/OpensearchContainer.java

```diff
@@ -0,0 +1,172 @@
+        this(DockerImageName.parse(dockerImageName));
+    }
+
+    /**
+     * Create an Opensearch Container (with security plugin enabled) by passing the full docker image
+     * name.
+     *
+     * @param dockerImageName Full docker image name as a {@link DockerImageName}, like:
+     *
+     *      DockerImageName.parse("opensearchproject/opensearch:1.2.4")
+     *      DockerImageName.parse("opensearchproject/opensearch:1.3.1")
+     *      DockerImageName.parse("opensearchproject/opensearch:2.0.0")
+     *
+     */
+    public OpensearchContainer(final DockerImageName dockerImageName) {
+        this(dockerImageName, false);
+    }
+
+    /**
+     * Create an Opensearch Container by passing the full docker image name.
+     *
+     * @param dockerImageName Full docker image name as a {@link DockerImageName}, like:
+     *
+     *    DockerImageName.parse("opensearchproject/opensearch:1.2.4")
+     *    DockerImageName.parse("opensearchproject/opensearch:1.3.1")
+     *    DockerImageName.parse("opensearchproject/opensearch:2.0.0")
+     *
+     * @param disableSecurity Should the security plugin be enabled or disabled. If the security
+     *     plugin is enabled, HTTPS protocol is going to be used along with the default username /
+     *     password.
+     */
+    public OpensearchContainer(final DockerImageName dockerImageName, boolean disableSecurity) {
+        super(dockerImageName);
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+        this.useHttps = !disableSecurity;
+
+        withNetworkAliases("opensearch-" + Base58.randomString(6));
+        withEnv("discovery.type", "single-node");
+        if (disableSecurity) {
+            withEnv("DISABLE_SECURITY_PLUGIN", Boolean.toString(disableSecurity));
+        }
+        addExposedPorts(DEFAULT_HTTP_PORT, DEFAULT_TCP_PORT);
+
+        final WaitStrategy waitStrategy;
+        if (useHttps) {
+            // By default, Opensearch uses self-signed certificates for HTTPS, allowing insecure
+            // connection in order to skip the certificate validation checks.
+            waitStrategy = new HttpWaitStrategy()
+                    .usingTls()
+                    .allowInsecure()
+                    .forPort(DEFAULT_HTTP_PORT)
+                    .withBasicCredentials(DEFAULT_USER, DEFAULT_PASSWORD)
+                    .forStatusCodeMatching(response -> response == HTTP_OK || response == HTTP_UNAUTHORIZED)
+                    .withReadTimeout(Duration.ofSeconds(10))
+                    .withStartupTimeout(Duration.ofMinutes(5));
+        } else {
+            waitStrategy = new HttpWaitStrategy()
+                    .forPort(DEFAULT_HTTP_PORT)
+                    .forStatusCodeMatching(response -> response == HTTP_OK)
+                    .withReadTimeout(Duration.ofSeconds(10))
+                    .withStartupTimeout(Duration.ofMinutes(5));
+        }
+
+        setWaitStrategy(waitStrategy);
```

Comment:
```suggestion
        waitStrategy = new HttpWaitStrategy()
            .forPort(DEFAULT_HTTP_PORT)
            .forStatusCodeMatching(response -> response == HTTP_OK)
            .withReadTimeout(Duration.ofSeconds(10))
            .withStartupTimeout(Duration.ofMinutes(5));
        
        if (useHttps) {
            // By default, Opensearch uses self-signed certificates for HTTPS, allowing insecure
            // connection in order to skip the certificate validation checks.
            ((HttpWaitStrategy) waitStrategy)
                .usingTls()
                .allowInsecure()
                .withBasicCredentials(DEFAULT_USER, DEFAULT_PASSWORD);
        }
```

Just as a suggestion 🙂

Reply:
Not typecasting was one of the reasons to have that previous branching logic, it may not be necessary anymore in case of https://github.com/opensearch-project/opensearch-testcontainers/pull/2#discussion_r936382547